sentry-spring/src/test/kotlin/io/sentry/spring/SentryUserProviderEventProcessorTest.kt

```diff
@@ -37,12 +37,13 @@ class SentryUserProviderEventProcessorTest {
         val result = processor.process(event, null)
 
         assertNotNull(result)
-        assertNotNull(result.user)
-        assertEquals("john.doe", result.user.username)
-        assertEquals("user-id", result.user.id)
-        assertEquals("192.168.0.1", result.user.ipAddress)
-        assertEquals("john.doe@example.com", result.user.email)
-        assertEquals(mapOf("key" to "value"), result.user.others)
+        assertNotNull(result.user) {
+            assertEquals("john.doe", it.username)
+            assertEquals("user-id", it.id)
+            assertEquals("192.168.0.1", it.ipAddress)
+            assertEquals("john.doe@example.com", it.email)
+            assertEquals(mapOf("key" to "value"), it.others)
+        }
```

Comment:
this block is duplicated across tests, maybe we extract it to a method called asserUser or something like that?

Reply:
I think in case of this test class duplicated code is a good trade off. Having fixed string values in `assertUser` method although resolves duplication, doesn't feel right to couple this method with test methods. Feel free to refactor it if you like.

Reply:
tbh I'm not a big fan of that, it's the same strings and same asserts across every test, every time you'd need to change either a value or adding/removing a field, you've to change it everywhere.
I know @philipphofmann has strong opinions about that too, it's not scalable when the number of tests increases.

Reply:
I agree 100% with @marandaneto. Treat your test code as production code. A few duplications are fine, but in this case, I would extract them to a method.

Reply:
I will merge as it is because I am blocked by this PR, but lets please discuss and refactor this as soon as we touch this class again

Reply:
I'm not willing to die on this hill so bare with me:

I believe a test should be as self-sufficient as possible. When a test calls out to a private method `getUser` then `assertUser` it isn't clear what `user` is, and how `assert` works, unless I peek into those methods. It also means that I can be setting the user now thinking of a particular test, but be negatively affecting another, without knowing, since multiple tests "change" by means of modifying these common functions.

It has the advantage, obviously, of having a single place to "fix" things, or even to add a new property to be set and asserted. It  But that comes with the cost of potentially changing a test that wasn't supposed to be changed.

Perhaps what we're missing here is a library that validates two objects are "equivalent". It solves the problem of adding a new property to the field and also expecting that to be mapped through which is a common mistake when mapping objects by hand.

Both these principles/approaches are used in the .NET SDK. The assertion library (introduced by @Tyrrrz, who from what I noticed also follows the idea of self-sufficient tests) is called `FluentAssertions `, is there something similar for Java?

All of that said, this class in particular does have a lot of duplication for basic object mapping, to the point that this diff was mainly changes to asserting `user`. I'd be very tempted to at least create a `assertUserEquals(User a, User b)` to get rid of this duplication.

Reply:
Kotlin generates `equals` and `hashCode` for data classes, but Java does not, we'd need to implement and bloat the SDK with that (which is gonna be used only in tests), so we could do an equality check.

another approach is having a test per field, instead of the whole user every time.

I do agree that's a trade-off, but I often spend more time scaling the tests other than "understanding" them.